qupulse/program/waveforms.py

```diff
@@ -1280,3 +1281,71 @@ def reversed(self) -> 'Waveform':
 
     def __repr__(self):
         return f"ReversedWaveform(inner={self._inner!r})"
+
+
+class WaveformCollection():
```

Comment:
Do we need WaveformCollection in the main repo?

I thnk it needs a different name and it belongs into a different file. I understand what the class is used for but I do not have a concept of what it actually represents (in terms of nesting). I think it is useful for general command table like devices, but needs more refinement before putting it into the main repo.